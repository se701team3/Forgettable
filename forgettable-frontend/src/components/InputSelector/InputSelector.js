import React, {useEffect} from 'react';
import {useState} from 'react';
import styles from './InputSelector.module.css';

/**
 * React input selector component used for text entry
 *
 * The styling of the component is based on that of the InputField component ("primary")
 *
 * Function Arguments:
 * @param inputLabel refers to the label associated with the text input component.
 * @param inputType refers to the "inputType" parameter as mentioned above.
 * @param placeholder refers to the placeholder text in the component.
 * @param dataType refers to the data format allowed as input.
 * @param inputID refers to the ID for the data associated with the input.
 * @param inputStateValue refers to the pre-filled value when the input is rendered.
 * @param autoFocusState refers to the autoFocus attribute, parameter is a Boolean.
 * @param requiredState refers to the required attribute, parameter is a Boolean.
 * @param options refers to an array of the options to be displayed in the dropdown.
 *
 * @author Bill Song
 */

export default function InputField({inputLabel, placeholder,
  dataType, inputID, inputStateValue, autoFocusState, requiredState, options}) {
  /**
     * Retrieve state of input component to be rendered in a responsive manner.
     */
  const [inputState, setInput] = useState(inputStateValue);

  useEffect(() => {
    setInput(inputStateValue);
  }, [inputStateValue]);

  /**
   *  process options into DOM elements
   */

  const optionElements = options.map((option) => {
    return <option key={option} value={option}>{option}</option>;
  });

  /**
     * Types allowed for innput component styling.
     */

  return (
    <div>
      <label className={styles.inputLabelPrimary}>{inputLabel}
        <select className={styles.inputFieldPrimary}
          type={dataType}
          name={inputID}
          placeholder = {placeholder}
          value={inputState}
          onChange={(event) => setInput(event.target.value)}
          autoFocus={autoFocusState}
          required={requiredState}
        >
          {optionElements}
        </select>
      </label>
    </div>
  );
}
