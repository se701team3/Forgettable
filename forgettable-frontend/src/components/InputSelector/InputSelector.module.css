@import '../../colours.css';

.inputLabelPrimary {
    font-size: var(--text-medium);
    font-weight: var(--font-medium);
    text-align: left;
    display: flex;
    flex-direction: column;
}

.inputFieldPrimary{
    font-size: var(--text-normal);
    background-color: var(--bkgd);
    color: var(--txt2);
    width: 370px;
    height: 57px;
    border-radius: var(--radius-btn);
    border: 1.5px solid var(--bord);
    box-shadow: none;
    margin-top: 8px;
    padding: 10px;
    padding-left: 18px;
    font-family: 'Poppins', sans-serif;
}
