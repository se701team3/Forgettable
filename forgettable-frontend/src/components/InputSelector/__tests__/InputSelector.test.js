import React from 'react';
import InputSelector from '../InputSelector';
import {render, cleanup} from '@testing-library/react';
import testRenderer from 'react-test-renderer';

afterEach(cleanup);

it('renders primary input field correctly', () => {
  const primaryInput = testRenderer.create(
      <InputSelector inputLabel={'SELECTOR'}
        placeholder={'SELECTOR TEXT'}
        dataType={'text'}
        inputID={'InputSelector'}
        options={['option1', 'option2', 'option3']} />).toJSON();
  expect(primaryInput).toMatchSnapshot();
});


