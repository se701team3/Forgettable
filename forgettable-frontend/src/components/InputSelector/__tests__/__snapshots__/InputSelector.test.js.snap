// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders primary input field correctly 1`] = `
<div>
  <label
    className="inputLabelPrimary"
  >
    SELECTOR
    <select
      className="inputFieldPrimary"
      name="InputSelector"
      onChange={[Function]}
      placeholder="SELECTOR TEXT"
      type="text"
    >
      <option
        value="option1"
      >
        option1
      </option>
      <option
        value="option2"
      >
        option2
      </option>
      <option
        value="option3"
      >
        option3
      </option>
    </select>
  </label>
</div>
`;
